pipenv/project.py

```diff
@@ -1,5 +1,3 @@
-from __future__ import annotations
```

Comment:
I think let's not remove this and the `if is_type_checking():` block below, since those are only needed for the mypy checking.

Reply:
This is needed if type hints are used in the code. Otherwise, there is NameError raised.
E.G, If I re-enable the guard, this line 
https://github.com/pypa/pipenv/blob/16b82949ecb2b3dd6fc18a7e79532ea6182eba67/pipenv/project.py#L266
will explode.
See the failed actions log here:

https://github.com/pypa/pipenv/actions/runs/3233835071/jobs/5296208253
![image](https://user-images.githubusercontent.com/1083045/196130291-2e3b536d-df93-422f-8f94-6020da85d24b.png)

This guard was really a must when supporting for Python2 was required. However, right now, it is not necessary.

Reply:
That's because List needs to be imported from collections.   I think we should not be undoing the type annotations.  I have a ticket to get my checks working again.

Reply:
It only List depends on being available outside of the annotations.  That is probably it.

Reply:
I re-introduced the guard, and will do a separate PR to remove this guard in  all the places we use it later.